src/unitxt/standard.py

```diff
@@ -153,19 +158,26 @@ def prepare(self):
         self.test_refiner.apply_to_streams = ["test"]
         self.steps.append(self.test_refiner)
 
-        render = StandardRenderer(
-            instruction=self.instruction,
-            template=self.template,
-            format=self.format,
-            demos_field=self.demos_field,
+        # the 4 constituents of StandardRenderer:
+        self.steps.append(self.template)
+        self.steps.append(
+            RenderDemonstrations(template=self.template, demos_field=self.demos_field)
         )
-
-        self.steps.append(render)
+        self.steps.append(
+            AddFields(
```

Comment:
Same question as above: Why set the instruction here on the instance, and not have it as a parameter to ModelInputFormatter?

Reply:
The formatter was already created (usually fetched from the catalog), so we can not pass the instruction in the constructor.   We have two options, add a method update_instruction() to update the object, or pass the instruction in the input stream (as is done now).  Note these "instructions" are really the system prompt.

Reply:
The former is more explicit, hence preferred IMO.

Generally, not being able to pass params to artifacts is a limitation .. what if we were just creating the formats with some function, and calling that from the card?